tests/ui/internal-lints/rustc_intrinsic_const_vector_arg.rs

```diff
@@ -0,0 +1,66 @@
+// compile-flags: -Z unstable-options
+#![feature(abi_unadjusted)]
+#![feature(inline_const)]
+#![feature(intrinsics)]
+#![allow(non_camel_case_types)]
+#![feature(portable_simd)]
+#![feature(repr_simd)]
+#![feature(rustc_attrs)]
+#![feature(simd_ffi)]
+#![allow(unused)]
+
+use std::simd::prelude::Simd;
+
+#[repr(simd)]
+#[derive(Clone)]
+pub struct i8x2(i8, i8);
+
+extern "unadjusted" {
+    #[rustc_intrinsic_const_vector_arg] //~ ERROR malformed `rustc_intrinsic_const_vector_arg` attribute input
+    fn foo1(a: i8x2, b: i8);
+}
+
+extern "unadjusted" {
+    #[rustc_intrinsic_const_vector_arg = "1"] //~ ERROR malformed `rustc_intrinsic_const_vector_arg` attribute input
+    fn foo2(a: i8x2, b: i8);
+}
+
+#[rustc_intrinsic_const_vector_arg(0)] //~ ERROR  attribute should be applied to functions in `extern "unadjusted"` modules
```

Comment:
Is there a reason this argument needs to be about const *vectors*?

If this were to become a general `rustc_intrinsic_const_arg`, then it could also be used for simd_insert/extract.